api/proto/teleport/legacy/types/types.proto

```diff
@@ -2855,6 +2855,22 @@ enum CreateDatabaseUserMode {
   DB_USER_MODE_BEST_EFFORT_DROP = 3;
 }
 
+// SSHPortForwardConfig defines which types of SSH port forwarding are permitted, if any.
+message SSHPortForwardConfig {
+  // Allow local port forwarding.
+  BoolValue Local = 1 [
```

Comment:
They can. it was just to avoid having to deprecate another field and keep supporting it because we never know when/how customers upgrade their clusters.

Reply:
I'm actually combining these bools into a single enum in the implementation, so I could definitely see lifting that into the proto instead. The only thing I don't really like about proto enums is that they're not very nice to work with from the terraform provider since they're effectively magic numbers. That's less of a problem than the schema being inflexible, though

Reply:
If you end up changing to enums, please use plain strings.
Otherwise, we end up with this (as you pointed out)

![image](https://github.com/user-attachments/assets/0e8b5ce0-f2ab-4baa-bd31-70f03905c3f9)

https://goteleport.com/docs/reference/terraform-provider/resources/auth_preference/

Reply:
After mulling it over some more and talking to TIm, I think leaving them as bools might make for a better user experience. Most of the extensions I can imagine we might want to do would probably involve extending the `SSHPortForwardConfig` message to contain more fields rather than modifying `Local` or `Remote` directly.

Reply:
I can see this evolving in a way that we would list the allowed ports.
Similar to TCPPorts in the App Resource: https://github.com/gravitational/teleport/pull/47706

I would prefer something more flexible, but I'm ok with the current design 👍

Reply:
I suppose the `Local` and `Remote` fields could be messages themselves, which would allow us to do something like:
```yaml
ssh_port_forwarding:
  local:
    enabled: true
    ports:
      - 8080
  remote:
    enabled: false
```
It's a little more verbose, but it saves us from having to add top level fields like `local_ports` and `remote_ports` later on 🤔 I'll think about this some more, but I might like that better